jerry-core/mem/mem-allocator.cpp

```diff
@@ -53,26 +53,7 @@ mem_finalize (bool is_show_mem_stats) /**< show heap memory stats
 
   if (is_show_mem_stats)
   {
-    mem_heap_print (false, false, true);
-
-#ifdef MEM_STATS
-    mem_pools_stats_t stats;
-    mem_pools_get_stats (&stats);
-
-    printf ("Pools stats:\n");
-    printf (" Chunk size: %zu\n"
-            "  Pools: %zu\n"
-            "  Allocated chunks: %zu\n"
-            "  Free chunks: %zu\n"
-            "  Peak pools: %zu\n"
-            "  Peak allocated chunks: %zu\n\n",
-            MEM_POOL_CHUNK_SIZE,
-            stats.pools_count,
-            stats.allocated_chunks,
-            stats.free_chunks,
-            stats.peak_pools_count,
-            stats.peak_allocated_chunks);
-#endif /* MEM_STATS */
+    mem_stats_print ();
```

Comment:
@ruben-ayrapetyan I've been considering that as well. The reason why I proposed the above is that it remained semantically equivalent with the original version, which also left the call to `mem_heap_print` unguarded. Even in `mem_heap_print`, there are quite some code lines which are always compiled, and only a final portion is ifdef'd by `MEM_STATS`. If we choose to ifdef the call here, then maybe we should consider putting much more code under ifdef guards,